packages/template-chakra-storefront/src/hooks/use-wish-list.js

```diff
@@ -5,17 +5,43 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {useEffect} from 'react'
-import {useCustomerProductLists, useShopperCustomersMutation} from '@salesforce/commerce-sdk-react'
-import {useCurrentCustomer} from './use-current-customer'
+import React, {useEffect, useCallback} from 'react'
+import {useIntl} from 'react-intl'
+import {Button} from '@chakra-ui/react'
+import {
+    useCustomerProductLists,
+    useShopperCustomersMutation,
+    useCustomerId
+} from '@salesforce/commerce-sdk-react'
+import useToast from './use-toast'
+import useNavigation from './use-navigation'
+import {
+    API_ERROR_MESSAGE,
+    TOAST_ACTION_VIEW_WISHLIST,
+    TOAST_MESSAGE_ADDED_TO_WISHLIST,
+    TOAST_MESSAGE_ALREADY_IN_WISHLIST,
+    TOAST_MESSAGE_REMOVED_FROM_WISHLIST
+} from '../../config/constants'
 
 const onClient = typeof window !== 'undefined'
-// TODO: remove `listId` input -> use the first list of type wish_list instead
-// (mimic the logic in the other older hook 'use-wishlist.js')
+
+const normalizeProductId = (product, variant) => {
+    return variant?.productId || product?.productId || product?.id
+}
```

Comment:
Great working going to extra mile to clean up and re-use a lot of code. Following that theme, do you think that a this utility function will be something that is useful elsewhere, no or in the future? 

E.g. do we want to move this to a utils file, export it, which will directly allow us to write unit tests.